database/rekordbox_pdb.ksy

```diff
@@ -390,19 +390,34 @@ types:
           and linked from other rows (such as tracks).
       - type: u1
         doc: |
-          @flesniak says: "alwayx 0x03, maybe an unindexed empty string"
+          @flesniak says: "always 0x03, maybe an unindexed empty string"
       - id: ofs_name
         type: u1
         doc: |
           The location of the variable-length name string, relative to
-          the start of this row.
+          the start of this row, unless subtype ix 0x64.
     instances:
       name:
         type: device_sql_string
+        if: subtype == 0x60
         pos: _parent.row_base + ofs_name
         doc: |
           The name of this artist.
         -webide-parse-mode: eager
+      ofs_long_name:
+        type: u2
+        if: subtype == 0x64
+        pos: _parent.row_base + 0x0a
```

Comment:
```yaml
    ofs_real_name:
        value: _parent.row_base + (subtype == 0x64 ? ofs_long_name : ofs_name)
    name:
        type: device_sql_string
        if: subtype == 0x60
        pos: ofs_real_name
```

Reply:
Also pleas follow the conv3ntion `pos [io] type desc`

Reply:
Around the time you were writing this, I was at my lunchtime swim, and in the pool is often when I review recent code and make improvements and optimizations, and I came up with the same idea, it will make dealing with the value much easier. Thanks! I will fix the order of the fields as well.

Reply:
Rats, when I try that I am getting a KSC error in the generate phase: `mapping values are not allowed here  in 'reader', line 410, column 66:  ... (subtype == 0x64)? ofs_name_far : ofs_name_near)` (column 66 is right after the `:`. Any ideas?

Reply:
Ah sorry, that’s a FAQ, need to escape it as a string for the silly YAML format.

Reply:
I ended up doing it a slightly different way, but it has the same net effect, the struct is easier to understand and _much_ easier to use; the name is always accessed using the same instance regardless of how it is actually stored.
https://github.com/brunchboy/kaitai_struct_formats/commit/a2898a03a65db4f1f7e19f7e3bf84d16ba338f65